provider/views.py

```diff
@@ -511,46 +551,12 @@ def error_response(self, error, status=400, **kwargs):
         kwargs.setdefault('content_type', 'application/json')
         return HttpResponse(json.dumps(error), status=status, **kwargs)
 
-    def access_token_response_data(self, access_token):
-        """
-        Returns access token data as defined in :rfc:`5.1`.
-
-        Derived classes can override to add extra parameters.
-        """
-
-        response_data = {
-            'access_token': access_token.token,
-            'token_type': constants.TOKEN_TYPE,
-            'expires_in': access_token.get_expire_delta(),
-            'scope': ' '.join(scope.names(access_token.scope)),
-        }
-
-        # Not all access_tokens are given a refresh_token
-        # (for example, public clients doing password auth)
-        try:
-            rt = access_token.refresh_token
-            response_data['refresh_token'] = rt.token
-        except ObjectDoesNotExist:
-            pass
-
-        return response_data
-
     def access_token_response(self, access_token, data=None):
         """
         Returns a successful response after creating the access token
         as defined in :rfc:`5.1`.
         """
         response_data = self.access_token_response_data(access_token)
-
-        if data is not None and data.get('response_type') == 'token':
-            basepath = data.get("redirect_uri")
-            if not basepath:
-                basepath = access_token.client.redirect_uri
-            if len(data.get('state', '')) > 0:
-                response_data['state'] = data.get('state')
-            path = "{0}#{1}".format(basepath, urlencode(response_data))
-            return HttpResponseRedirect(path)
```

Comment:
This code was added on the AccessToken view recently, specifically to support implicit flow, which is signified by `response_type='token'` in the authorization request.  Part of this cleanup was to move this logic into `get_implicit_response` because it actually doesn't make sense to have it in the AccessToken view (because 'response type' is not a valid parameter for access token requests).   (short answer - yes, it is no longer needed)